src/people.rs

```diff
@@ -553,4 +775,129 @@ mod test {
+        let mut context = Context::new();
+        let person = context.add_person();
+        context.initialize_person_property(person, Age, 17);
+        assert_eq!(
+            context.get_person_property(person, AgeGroup),
+            AgeGroupType::Child
+        );
+        context.set_person_property(person, Age, 18);
+        assert_eq!(
+            context.get_person_property(person, AgeGroup),
+            AgeGroupType::Adult
+        );
+    }
+    #[test]
+    fn get_person_property_change_event() {
+        let mut context = Context::new();
+        let person = context.add_person();
+        context.initialize_person_property(person, Age, 17);
+
+        let flag = Rc::new(RefCell::new(false));
+        let flag_clone = flag.clone();
+        context.subscribe_to_event(
+            move |_context, event: PersonPropertyChangeEvent<AgeGroup>| {
+                assert_eq!(event.person_id.id, 0);
+                assert_eq!(event.previous, AgeGroupType::Child);
+                assert_eq!(event.current, AgeGroupType::Adult);
+                *flag_clone.borrow_mut() = true;
+            },
+        );
+        context.set_person_property(person, Age, 18);
+        context.execute();
+        assert!(*flag.borrow());
+    }
+
+    #[test]
+    fn get_derived_property_multiple_deps() {
+        let mut context = Context::new();
+        let person = context.add_person();
+        context.initialize_person_property(person, Age, 29);
+        context.initialize_person_property(person, IsRunner, true);
+
+        let flag = Rc::new(RefCell::new(false));
+        let flag_clone = flag.clone();
+        context.subscribe_to_event(
+            move |_context, event: PersonPropertyChangeEvent<TrailRunner>| {
+                assert_eq!(event.person_id.id, 0);
+                assert!(!event.previous);
+                assert!(event.current);
+                *flag_clone.borrow_mut() = true;
+            },
+        );
+        context.set_person_property(person, Age, 30);
+        context.execute();
+        assert!(*flag.borrow());
+    }
+
+    #[test]
+    fn register_derived_only_once() {
+        let mut context = Context::new();
+        let person = context.add_person();
+        context.initialize_person_property(person, Age, 29);
+        context.initialize_person_property(person, IsRunner, true);
+
+        let flag = Rc::new(RefCell::new(0));
+        let flag_clone = flag.clone();
+        context.subscribe_to_event(
+            move |_context, _event: PersonPropertyChangeEvent<TrailRunner>| {
+                *flag_clone.borrow_mut() += 1;
+            },
+        );
+        context.subscribe_to_event(
+            move |_context, _event: PersonPropertyChangeEvent<TrailRunner>| {
+                // Make sure that we don't register multiple times
+            },
+        );
+        context.set_person_property(person, Age, 30);
+        context.execute();
+        assert_eq!(*flag.borrow(), 1);
+    }
```

Comment:
Can you add a test with a diamond dependency that verifies that something only fires once: i.e.

A depends on B and C and B and C both depend on D?